### PR TITLE
better fix for extrafields

### PR DESCRIPTION
**Pull request by altairis-tof**
better fix for https://github.com/Dolibarr/dolibarr/pull/18102#event-4995386237

### CODE REVIEW

File: htdocs/core/class/commonobject.class.php [hunk 1]
```diff
@@ -7492,10 +7492,8 @@ public function showOptionals($extrafields, $mode = 'view', $params = null, $key
 
 						if ($display_type == 'card') {
 							$out .= '<tr '.($html_id ? 'id="'.$html_id.'" ' : '').$csstyle.' class="valuefieldcreate '.$class.$this->element.'_extras_'.$key.' trextrafields_collapse'.$extrafields_collapse_num.(!empty($this->id)?'_'.$this->id:'').'" '.$domData.' >';
-							if ( ! empty($conf->global->MAIN_VIEW_LINE_NUMBER) ) {
-								$out .= '<td></td>';
-							}
-							$out .= '<td class="wordbreak';
+							if ( ! empty($conf->global->MAIN_VIEW_LINE_NUMBER) ) $out .= '<td class="';
+							else $out .= '<td class="wordbreak';
```
Comment (eldy) [inline htdocs/core/class/commonobject.class.php:7496]: What was wrong with previous code ?

**Comment by altairis-tof** [comment]
it was adding a td before any card extrafield.
the real problem is that shipping lines are using "card" parameter when calling showoptionals function.

**Comment by eldy** [comment]
> showoptionals

May be this is that the trouble. It should use "line" instead of "card". What happen if you fix by switching into "line" ?

**Comment by altairis-tof** [comment]
the "line" parameter generate a div and not a tr; so the div goes above the shipping lines table and not below the shipping line itself.

**Comment by altairis-tof** [comment]
@eldy i think this fix is important

**Comment by eldy** [comment]
I have some reason to think this may be fixed into more recent version. Does this bug still occurs with last stable version ?
If not, can you close ticket ?
If yes, can you confirm the last stable version you reproduced the bug with ?

**Comment by altairis-tof** [comment]
@eldy yes the bug is still relevant (copac) and this PR fix it.

**Pull request closed without merging**